ml_tools/dataset.py

```diff
@@ -615,21 +660,28 @@ def load_tracks(self):
                 counter += 1
         return [counter, len(track_ids)]
 
-    def add_tracks(self, tracks):
+    def add_tracks(self, tracks, max_segments_per_track=None):
         """
         Adds list of tracks to dataset
         :param tracks: list of TrackHeader
         :param track_filter: optional filter
         """
         result = 0
         for track in tracks:
-            if self.add_track_header(track):
+            if self.add_track_header(track, max_segments_per_track):
                 result += 1
         return result
 
-    def add_track_header(self, track_header):
+    def add_track_header(self, track_header, max_segments_per_track=None):
         if track_header.track_id in self.track_by_id:
             return False
+
```

Comment:
Taking random track segments could mean that the segments still overlap I think (as from memory segments can overlap some of the frames) - don't worry about fixing that now.

Reply:
That is true, i wasn't worried so much about that.In the last build I increased the segment spacing to 2 seconds. 

I'm not sure if we need to be worried about having too many segments from a track. if we have enough data. But a whole bunch of work could be doing to improving the distribution.